components/gitpod-db/src/typeorm/entity/db-code-sync-resource.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { Entity, Column, PrimaryColumn, Index } from "typeorm";
+import { TypeORM } from "../typeorm";
+
+// should be aligned with https://github.com/gitpod-io/vscode/blob/75c71b49cc25554adc408e63b876b76dcc984bc1/src/vs/platform/userDataSync/common/userDataSync.ts#L113-L156
+export interface IUserData {
+    ref: string;
+    content: string | null;
+}
+
+export const enum SyncResource {
+    Settings = 'settings',
+    Keybindings = 'keybindings',
+    Snippets = 'snippets',
+    Extensions = 'extensions',
+    GlobalState = 'globalState'
+}
+export const ALL_SYNC_RESOURCES: SyncResource[] = [SyncResource.Settings, SyncResource.Keybindings, SyncResource.Snippets, SyncResource.Extensions, SyncResource.GlobalState];
+
+export interface IUserDataManifest {
+    latest?: Record<ServerResource, string>
+    session: string;
+}
+
+export type ServerResource = SyncResource | 'machines';
+export const ALL_SERVER_RESOURCES: ServerResource[] = [...ALL_SYNC_RESOURCES, 'machines'];
+
+@Entity()
+@Index('ind_dbsync', ['created'])   // DBSync
+export class DBCodeSyncResource {
```

Comment:
this data needs to go through db sync. I.e. you'll need a follow-up PR in com to add it to the `tables.ts`

Reply:
so, it is not enough to add them here: https://github.com/gitpod-io/gitpod/pull/3110/files#diff-808d43ade9932437a2de01c46e468fe131ec3b6cb8a943c7506c46d421e518fe ?